packages/Animation/FadeAndReveal.jsx

```diff
@@ -8,31 +8,40 @@ export const StyledContainer = styled.div({
     transition: 'none !important',
   },
 })
-const defaultStyle = (duration, timeout, state) => ({
+const defaultStyle = (duration, timeout) => ({
   transition: `height ${duration || timeout}ms, opacity ${duration || timeout}ms ease-in-out`,
   opacity: 0,
   height: 0,
   overflow: 'hidden',
-  visibility: state === 'exited' || state === 'exiting' ? 'hidden' : 'visible',
 })
 
 const transitionStyles = height => (duration, timeout) => ({
-  entering: { opacity: 1, height },
-  entered: { opacity: 1, height: 'auto' },
+  entering: {
+    opacity: 1,
+    height,
+    visibility: 'visible',
+    transition: `height ${duration || timeout}ms, opacity ${duration ||
+      timeout}ms ease-in-out, visibility 0s`,
```

Comment:
Can you transition `visibility`?

Reply:
Yes and no. In this case, we use a delay, which does work.